x-pack/plugins/actions/server/builtin_action_types/lib/connector_token_client.ts

```diff
@@ -245,4 +252,36 @@ export class ConnectorTokenClient {
       throw err;
     }
   }
+
+  public async updateOrReplace({
```

Comment:
This logic was taking place inside the email executor and needed to be duplicated for ServiceNow. I extracted this logic into its own function for reuse.